### DISCUSSION TITLE
Dataset '{}' is empty! Error (While train custom dataset on detectron2)

### DISCUSSION THREAD

**Discussion by MuharremcanGulye** [facebookresearch/detectron2#5093] [Q&A]
Hello all,
First of all i am almost a newbie.
I am trying to train a model that detects objects on Detectron2.
Problem is all my annotations are csv format and im trying to convert them proper format.
Im trying to imitate baloon detection tutorial.

Here is my code,

import numpy as np
import os, json, cv2, random
import pandas as pd
import cv2

def getdict(path):
  files = os.listdir(path)

  categories = {'cat1':1, 'cat2: 2, 'cat3':3 }
 
  for file in files:
    image_id = int(file.split(".")[0])
    dataset_dicts = []
    if file[-4:] == '.jpg':
      record = {}

      filename = os.path.join(path,file)
      height, width = cv2.imread(filename).shape[:2]

      record["file_name"] = file
      record["image_id"] = image_id
      record["height"] = height
      record["width"] = width

      annotation_file_name = file.split('.')[0]
      annotation_file_path = os.path.normpath(annotation_file_name + '.csv')

      annotation_path = os.path.join(path,annotation_file_path)

      df = pd.read_csv(annotation_path)

      for idx, row in df.iterrows():
        idx += 1
        objs = []
        category = row["raw_classification"]
        obj = {
          "bbox": [row['xmin'], row['ymin'], row['xmax'], row['ymax']],
          "bbox_mode": "BoxMode.XYXY_ABS",
          "category_id" : categories[category]
        }
        objs.append(obj)
        record['annotations'] = objs
        print(record)
        dataset_dicts.append(record)
getdict("/content/drive/MyDrive/test1/")

The error that i get is down below.

[/content/detectron2/detectron2/data/build.py](https://localhost:8080/#) in get_detection_dataset_dicts(names, filter_empty, min_keypoints, proposal_files, check_consistency)
    262 
    263     for dataset_name, dicts in zip(names, dataset_dicts):
--> 264         assert len(dicts), "Dataset '{}' is empty!".format(dataset_name)
    265 
    266     if proposal_files is not None:

TypeError: object of type 'NoneType' has no len()

I think i am doing something wrong with creating dataset_dicts but i coudnt find what is wrong.

Thanks and regards.

**Comment by rdaggs** [comment]
This is an error I am having as well.....Training dataset loads fine, iterates through the epochs.....No such luck with validation dataset.

**Reply by hehongjie** [reply]
have you solved it?

**Comment by hehongjie** [comment]
have you solved the problem?

**Reply by rdaggs** [reply]
It was a file naming convention. Ensure paths are routed correctly.
